### ISSUE TITLE
Universal Message Signing with Collateral Verification for Multiple Blockchains

### ISSUE THREAD

**Issue by BFWookie** [vultisig/vultisig-ios#1288]
Description:

We need a comprehensive solution in Vultisig for message signing and collateral verification that works across multiple blockchains, not just Dash. This feature is crucial for various cryptocurrency operations, including masternode management.

Background:
Trezor is removing Dash support from Trezor Suite in February 2025, highlighting the need for alternative solutions. However, message signing is a universal feature beneficial for many blockchain networks.

Feature Request Details:

1. Implement a blockchain-agnostic message signing feature.
2. Provide functionality to confirm the exact public key and input index where collateral (or any specific UTXO) is located.
3. Allow users to choose which public key to use for signing when multiple options are available.
4. Offer a user-friendly interface for entering and signing custom messages.
5. Generate signatures compatible with multiple blockchain networks, not limited to Dash.
6. Include a verification feature to check the validity of signed messages.

User Story:
As a cryptocurrency user (including but not limited to Dash masternode owners), I want to use Vultisig to:
a) Confirm the exact location (public key and input index) of specific UTXOs
b) Choose the appropriate public key for signing
c) Sign messages to prove ownership of addresses or specific coins
So that I can securely manage my crypto assets and perform verifications across various blockchain networks.

Acceptance Criteria:
1. Users can view and confirm the public key and input index associated with specific UTXOs for multiple supported cryptocurrencies.
2. The wallet provides an option to select from available public keys when multiple are present.
3. Users can sign custom messages using the selected public key for any supported blockchain.
4. The signing process is secure and does not expose private keys.
5. Signed messages are compatible with standard wallets of respective blockchains.
6. Users can verify signed messages within the wallet for multiple cryptocurrencies.
7. The interface clearly displays which public key was used for signing.
8. The process is intuitive and provides clear guidance for users transitioning from other wallets, including Trezor Suite.

Additional Context:
While this feature is particularly urgent for Dash users due to upcoming Trezor changes, implementing it as a universal feature will benefit a wider range of cryptocurrency users. This approach ensures Vultisig's versatility and appeal across different blockchain communities.

Attached Examples:
Please refer to the attached screenshots from Trezor Suite, which demonstrate their implementation of message signing. These can serve as a reference for user interface design and functionality.

![Screenshot from 2024-10-08 10-08-50](https://github.com/user-attachments/assets/47149472-1c58-4e87-8a4f-24db4da8fee2)

![Screenshot from 2024-10-08 10-09-43](https://github.com/user-attachments/assets/d58f16a6-1387-4bee-b4a6-77985864862c)

![Screenshot from 2024-10-08 10-09-17](https://github.com/user-attachments/assets/74d2cd2a-e334-4631-b479-5dbdd70635a5)

These examples showcase:
1. The interface for entering custom messages
2. The process of selecting which address/public key to sign with
3. The display of signed messages and their verification

Labels: enhancement, security, cross-chain, message-signing, collateral-management

**Comment by johnnyluo** [comment]
Dash is already supported in Vultisig , but this issue is discussing a custom message signing , at this stage , we are unlikely to implement this feature , so closing

**Comment by johnnyluo** [comment]
The vultisig core is TSS , which is capable of signing any arbitrary messages , but currently we limit it to only sign information that can be represent using 

https://github.com/vultisig/commondata/blob/main/proto/vultisig/keysign/v1/keysign_message.proto

This is all about transaction , doesn't cover custom message  , we could extend it to support custom message if need to

In regards to multiple address derive from the same root public key , we didn't implement it , because it often confuse user

**Comment by BFWookie** [comment]
Yes we just need custom message signing. Multiple addresses from the same root public key is not necessary. I suppose you could have a vault with a single address just for your collateral.

Just as long as you can sign a message from the same public key your asset is in to prove ownership for masternode purposes and to prove to entities that you control a specific address.

**Comment by johnnyluo** [comment]
Thinking about it more , we will build this in the future , because another feature that we are planning will require it too.

**Comment by BFWookie** [comment]
Yes indeed that would be great. Any user who needs to verify ownership of an address for a centralised or decentralised service provider has to have the ability to sign a custom message from that address.

**Comment by johnnyluo** [comment]
![Image](https://github.com/user-attachments/assets/7796af0d-231a-4705-b98e-da53f0bab941)

we added custom message sign , but now is only using ETH key , maybe need to extend it to support others

**Comment by realpaaao** [comment]
Closing as stale — 18 months with no activity. Reopen if still relevant.
